ergo/metaculus.py

```diff
@@ -185,9 +199,45 @@ def get_submission_params(self, logistic_params: logistic.LogisticParams) -> Sub
         # (we belive that if you set the low higher than the value below [or if you set the high lower],
         # then the API will reject the prediction, though we haven't tested that extensively)
         low = max(distribution.cdf(0), 0.01) if self.low_open else 0
-        high = min(distribution.cdf(1), 0.99) if self.high_open else 1
+
+        # min high of low + 0.01 set based on API response for https://www.metaculus.com/api2/questions/3961/predict/ --
+        # {'prediction': ['high minus low must be at least 0.01']}"
+        high = max(min(distribution.cdf(1), 0.99),
+                   low + 0.01) if self.high_open else 1
         return SubmissionLogisticParams(clipped_loc, clipped_scale, low, high)
 
+    # Map normalized samples back to actual scale
+    def denormalize_samples(self, samples):
+        raise NotImplementedError("This should be implemented by a subclass")
+
+    @functools.lru_cache(None)
+    def community_dist_in_range(self):
+        # distribution on integers referencing 0...(len(self.prediction_histogram)-1)
+        df = pd.DataFrame(self.prediction_histogram, columns=["x", "y1", "y2"])
+        return dist.Categorical(probs=torch.Tensor(df["y2"]))
+
+    def sample_normalized_community(self):
+        sample_below_range = -1 * \
+            abs(np.random.logistic(0, 0.02))
+        sample_above_range = abs(np.random.logistic(
+            1, 0.02))
+        sample_in_range = ppl.sample(self.community_dist_in_range(
+        )) / float(len(self.prediction_histogram))
```

Comment:
Thanks, I like the idea of switching to Black and created an issue for it: [#80](https://github.com/oughtinc/ergo/issues/80)

Reply:
no missing argument I don't think -- note that we're using a different function to grab a sample than for the other 2

yeah I think self.prediction_histogram is always non-empty -- would probably rather that we just get an error if it's not so that we know that something unexpected happened (rather than handling that case)